perfkitbenchmarker/linux_benchmarks/cloudsuite_in_memory_analytics_benchmark.py

```diff
@@ -0,0 +1,159 @@
+  vm_groups:
+    master:
+      vm_spec: *default_single_core
+      vm_count: 1
+    workers:
+      vm_spec: *default_single_core
+"""
+
+
+def GetConfig(user_config):
+  config = configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+  if FLAGS['num_vms'].present:
+    config['vm_groups']['workers']['vm_count'] = FLAGS.num_vms
+  return config
+
+
+def _HasDocker(vm):
+  resp, _ = vm.RemoteCommand('command -v docker',
+                             ignore_failure=True,
+                             suppress_warning=True)
+  return bool(resp.rstrip())
+
+
+def Prepare(benchmark_spec):
+  """Install docker. Pull the required images from DockerHub. Create datasets.
+  Start Spark master and workers.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  master = benchmark_spec.vm_groups['master'][0]
+  workers = benchmark_spec.vm_groups['workers']
+
+  def PrepareCommon(vm):
+    if not _HasDocker(vm):
+      vm.Install('docker')
+    vm.RemoteCommand('sudo docker pull cloudsuite/spark')
+    vm.RemoteCommand('sudo docker pull cloudsuite/movielens-dataset')
+    vm.RemoteCommand('sudo docker create --name data '
+                     'cloudsuite/movielens-dataset')
+
+  def PrepareMaster(vm):
+    PrepareCommon(vm)
+    vm.RemoteCommand('sudo docker pull cloudsuite/in-memory-analytics')
+    start_master_cmd = ('sudo docker run -d --net host -e SPARK_MASTER_IP=%s '
+                        '--name spark-master cloudsuite/spark master' %
+                        master.internal_ip)
+    vm.RemoteCommand(start_master_cmd)
+
+  def PrepareWorker(vm):
+    PrepareCommon(vm)
+    start_worker_cmd = ('sudo docker run -d --net host --volumes-from data '
+                        '--name spark-worker cloudsuite/spark worker '
+                        'spark://%s:7077' % master.internal_ip)
+    vm.RemoteCommand(start_worker_cmd)
+
+  target_arg_tuples = ([(PrepareWorker, (vm,), {}) for vm in workers] +
+                       [(PrepareMaster, (master,), {})])
+  vm_util.RunParallelThreads(target_arg_tuples, len(target_arg_tuples))
+
+
+def Run(benchmark_spec):
+  """Run the in-memory analytics benchmark.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  master = benchmark_spec.vm_groups['master'][0]
+  results = []
+
+  benchmark_cmd = ('sudo docker run --rm --net host --volumes-from data '
+                   'cloudsuite/in-memory-analytics %s %s '
+                   '--master spark://%s:7077' %
+                   (FLAGS.cloudsuite_in_memory_analytics_dataset,
+                    FLAGS.cloudsuite_in_memory_analytics_ratings_file,
+                    master.internal_ip))
+  stdout, _ = master.RemoteCommand(benchmark_cmd, should_log=True)
+
+  execution_time = re.findall(r'Benchmark execution time: (\d+)ms', stdout)[0]
```

Comment:
I'm not really proficient at python, actually never used it before. What do you suggest here?